addons/assigngear/UI/TMF_AttributesFaction.sqf

```diff
@@ -0,0 +1,57 @@
+#include "..\script_component.hpp"
+
+disableSerialization;
+
+params ["_mode","_params","_unit"];
+_params params ["_ctrl","_config"];
+TRACE_3("Loading AIGear Faction Attribute",_mode,_params,_unit);
+
+#ifdef CURATOR_ATTRIBUTE
+    private _ctrlCombo = _ctrl displayctrl IDC_RSCATTRIBUTEFACTION_COMBO;
+#else
+    private _ctrlCombo = _ctrl controlsGroupCtrl 100;
+#endif
+
+switch (_mode) do {
+    case "onLoad": {
+        private _factionClasses = uiNamespace getVariable [QGVAR(factions),[]];
+        if (_factionClasses isEqualTo []) then {
+            LOG("AIGear factionClasses not cached");
+            // Cache to uiNamespace
+            _factionClasses = "isNumber (_x >> 'side') && getNumber (_x >> 'side') in [0,1,2,3]" configClasses (configFile >> 'CfgFactionClasses');
+
+            // Filter to only factions that have units
+            _factionClasses = _factionClasses select {
+                private _cfgName = configName _x;
+                private _men = "configName _x isKindOf 'CAManBase'" configClasses (configFile >> "CfgVehicles");
+                (_men findIf {getText (_x >> "faction") == _cfgName}) != -1;
+            };
+
+            _factionClasses = _factionClasses apply {[
+                configName _x,
+                getText (_x >> 'displayName') + [" [OPF]", " [BLU]", " [IND]", " [CIV]"] # getNumber (_x >> "side"),
+                getText (_x >> 'icon')
+            ]};
+            uiNamespace setVariable [QGVAR(factions),_factionClasses];
+            LOG_1("AIGear factionClasses cached to uiNamespace: %1",_factionClasses);
+        };
+
+
+        {
+            _x params ['_data','_displayName','_icon'];
+            TRACE_3("AIGear Adding to faction combo",_data,_displayName,_icon);
+            private _id = _ctrlCombo lbAdd _displayName;
+            _ctrlCombo lbSetData [_id,_data];
+            _ctrlCombo lbSetPictureRight [_id,_icon];
+        } forEach _factionClasses;
+        lbSort _ctrlCombo;
+        _ctrlCombo lbSetCurSel 0;
+    };
+    #ifdef CURATOR_ATTRIBUTE
+    case "onUnload": {};
+    case "confirmed": {
+        _unit setVariable ["Faction",_ctrlCombo lbData lbCurSel _ctrlCombo];
```

Comment:
This is a nitpick but generically named variables like "Faction" and "Updated" on a unit they could collide with other mods/scripts so using `GVAR` prefixes is advisable. As you pointed out to me this is a module. I'd suggest also renaming the `_unit` variable to be either `_module` or `_logic` which makes it easier for the reader. I tend to assume a variable named `_unit` would be a man object.

As it is a module other mods shouldn't be setting variables on it, so I'll settle for renaming the variable to `_logic` or `_module` (or something similiar)

Reply:
The `updated` part is simply following the BI curator module standard, which uses
```sqf
waituntil {!isnil {_logic getvariable "updated"} || isnull _logic};
```